src/state/actions/dispatchANTInteraction.ts

```diff
@@ -1,36 +1,54 @@
 import {
   ANT,
+  AOS_MODULE_ID as ANT_MODULE_ID,
   AOProcess,
   ARIO,
+  AoANTHandler,
   AoClient,
   AoMessageResult,
   ContractSigner,
   createAoSigner,
   spawnANT,
 } from '@ar.io/sdk/web';
+import { buildDomainInfoQuery } from '@src/hooks/useDomainInfo';
 import { TransactionAction } from '@src/state/reducers/TransactionReducer';
 import { ANT_INTERACTION_TYPES, ContractInteraction } from '@src/types';
 import { lowerCaseDomain, sleep } from '@src/utils';
+import { ARWEAVE_HOST, NETWORK_DEFAULTS } from '@src/utils/constants';
 import eventEmitter from '@src/utils/events';
 import { queryClient } from '@src/utils/network';
 import { Dispatch } from 'react';
 
+import { ArNSAction } from '../reducers';
+
 export default async function dispatchANTInteraction({
   payload,
   workflowName,
   processId,
   signer,
   owner,
-  dispatch,
+  dispatchTransactionState,
+  dispatchArNSState,
   ao,
+  // this can allow for waiting on promise resolution for UI input on individual steps
+  stepCallback = async (step) => {
+    if (typeof step === 'string') {
+      dispatchTransactionState({
```

Comment:
Unsure whether this gets the dispatchTransactionState from args here. If it does, it's still not as clear to have that as part of default arg here vs. creating the callback within the function body as it was being done before where it was creating dispatchSigningMessage. Would prefer the stepCallback function be overridden if undefined within the body.

Reply:
I moved it but i dont like it https://github.com/ar-io/arns-react/pull/704/commits/9de91f09f26cd63ca5be8528fc258dccf0dc4080